src/main/java/org/apache/accumulo/access/AccessEvaluator.java

```diff
@@ -149,7 +153,32 @@ static AccessEvaluator of(Authorizer authorizer) {
    *
    */
   static AccessEvaluator of(Collection<Authorizations> authorizationSets) {
-    return new AccessEvaluatorImpl(AccessEvaluatorImpl.convert(authorizationSets));
+    List<AccessEvaluatorImpl> evaluators = new ArrayList<>(authorizationSets.size());
+    for (Authorizations authorizations : authorizationSets) {
+      evaluators.add(new AccessEvaluatorImpl(authorizations));
+    }
+
+    return new AccessEvaluator() {
```

Comment:
If this returns an implementation of AccessEvaluator, do we still need AccessEvaluatorImpl? If so, do we end up having two different implementations?

Reply:
> If this returns an implementation of AccessEvaluator, do we still need AccessEvaluatorImpl? If so, do we end up having two different implementations?

Yeah there are two impls of the interface now.  Still having  AccessEvaluatorImpl makes it really easy for the multiple auths impl to use the single auths impl, so thinking it simplifies the code.

Reply:
Ok, I think I understand this PR better now. You have modified `AccessEvaluatorImpl` such that it only operates on a single `Authorizations` object and moved the function of dealing with multiples into this anonymous object. I have no issue with this, but wondering if we should/could move this anonymous class to a concrete class, call it `AccessEvaluatorsImpl`, `MultiAccessEvaluatorImpl`, or something like that.

Reply:
> but wondering if we should/could move this anonymous class to a concrete class, call it AccessEvaluatorsImpl, MultiAccessEvaluatorImpl, or something like that.

Yeah can do that, it would keep the interface file source code really compact and it makes stacktraces a bit more readable.